### PR TITLE
responsive header & footer Tailwind

### PR DESCRIPTION
**Pull request by Abd-alRahim-Kojan**

### CODE REVIEW

File: src/app/shared/page/footer/footer.component.html [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                >
+                  <img [src]="link.imgSrc" [alt]="link.imgAlt" class="w-64" />
+                </a>
+              </li>
+            </ul>
+
+            <ul class="flex flex-col gap-2.5">
+              <p class="text-xl font-bold text-teal-500">تابعنا</p>
+              <div
+                class="flex gap-2.5 flex pt-2 justify-center md:justify-start"
```
Comment (AMI2080) [inline src/app/shared/page/footer/footer.component.html]: remove duplicated class `flex`

File: src/app/shared/page/footer/footer.component.html [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  <div
+    class="w-full absolute top-0 left-0 right-0 w-full absolute inset-x-0 top-0 h-2 bg-gradient-to-r from-teal-500 to-blue-800"
+  >
+    <div class="bg-blue-950 mt-2">
+      <div class="bg-pattern bg-contain">
+        <footer class="container py-14">
+          <a
+            [routerLink]="['']"
+            class="flex items-center py-4 justify-center md:justify-start"
+            ><img
```
Comment (AMI2080) [inline src/app/shared/page/footer/footer.component.html]: start img tag in new line

File: src/app/shared/page/footer/footer.component.html [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+              src="assets/logo-white.svg"
+              alt="logo-icon-white"
+              class="w-16"
+            />
+            <div class="text-white font-black text-lg pr-2">
+              <P> SHOGOL </P>
+              <P>شـــغـــل</P>
+            </div>
```
Comment (AMI2080) [inline src/app/shared/page/footer/footer.component.html:18]: > this is part logo so the best is display it as img

> but keep it

File: src/app/shared/page/footer/footer.component.html [hunk 4]
```diff
@@ -0,0 +1,80 @@
+<div class="relative text-center">
+  <div
+    class="w-full absolute top-0 left-0 right-0 w-full absolute inset-x-0 top-0 h-2 bg-gradient-to-r from-teal-500 to-blue-800"
+  >
+    <div class="bg-blue-950 mt-2">
```
Comment (AMI2080) [inline src/app/shared/page/footer/footer.component.html]: we don't need to positions
```suggestion
<div class="text-center">
  <div class="h-2 bg-gradient-to-r from-teal-500 to-blue-800"></div>
  <div>
   <div class="bg-blue-950">
```

Comment (AMI2080) [reply src/app/shared/page/footer/footer.component.html]: do the same in header

File: src/app/shared/page/footer/footer.component.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  templateUrl: './footer.component.html',
+  styles: [
+    `
+      @media (min-width: 768px) {
+        .text-center {
+          text-align: right !important;
+        }
+      }
+    `,
+  ],
```
Comment (AMI2080) [inline src/app/shared/page/footer/footer.component.ts]: use class `text-center md:text-start` instead

File: src/app/shared/page/footer/footer.component.ts [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    { title: 'قدم كشريك', route: '/routePath' },
+    { title: 'تصفح كل الفئات', route: '/routePath' },
+    { title: 'اتصل بنا', route: '/routePath' },
+  ];
+
+  appLinks = [
+    {
+      imgSrc: 'assets/images/app-store.png',
+      imgAlt: 'app-store',
+      route: '/routePath',
```
Comment (AMI2080) [inline src/app/shared/page/footer/footer.component.ts]: use array routes

```suggestion
      route: ['', 'routePath'],
```

File: src/app/shared/page/header/header.component.html [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+                  isMenuOpen
+                    ? 'fa-solid fa-xmark fa-2xl transition-all'
+                    : 'fa-solid fa-bars fa-lg transition-all'
+                "
+              ></i>
+            </button>
+            <!-- Navigation links -->
+            <ul
+              class="flex flex-col md:flex-row md:gap-5 md:items-center md:z-auto md:static bg-white items-end w-full absolute left-0 p-6 pt-0 md:p-0 top-20 md:top-0"
+              [class]="isMenuOpen ? 'flex' : 'hidden'"
```
Comment (AMI2080) [inline src/app/shared/page/header/header.component.html]: use `[ngClass]` instead

```suggestion
              [ngClass]="isMenuOpen ? 'flex' : 'hidden'"
```

File: src/app/shared/page/header/header.component.ts [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+  templateUrl: './header.component.html',
+  styles: [
+    `
+      @media (min-width: 768px) {
+        .mt-3 {
+          margin-top: 0 !important;
+        }
+      }
+    `,
+  ],
```
Comment (AMI2080) [inline src/app/shared/page/header/header.component.ts]: again

we can use classes `mt-3 md:mt-0`

File: src/app/shared/shared.module.ts [hunk 9]
```diff
@@ -3,14 +3,25 @@ import { CommonModule } from '@angular/common';
...
+    NgSelectModule,
+    ReactiveFormsModule,
+    FormsModule,
+    RouterModule,
+  ],
   declarations: [
     // components
     ...sharedComponents,
+    HeaderComponent,
+    FooterComponent,
```
Comment (AMI2080) [inline src/app/shared/shared.module.ts]: add components to shared sharedComponents array in line 10

**Review by AMI2080 — COMMENTED** [review]
add data types to variables

File: src/app/shared/page/footer/footer.component.ts (hunk 5 above)
Comment (Abd-alRahim-Kojan) [reply src/app/shared/page/footer/footer.component.ts]: I tried it before and it did not work. I tracked the class and I think it's conflict from Bootstrap

File: src/app/shared/page/footer/footer.component.ts (hunk 6 above)
Comment (Abd-alRahim-Kojan) [reply src/app/shared/page/footer/footer.component.ts]: I tried it and it breaks the routes. it shows error. this is what I found about it

When you use an array for routing like ['', 'routePath'], Angular treats it as a relative path and tries to resolve it based on the current route. The empty string '' in the array is interpreted as the current route, and 'routePath' is appended to it.

File: src/app/shared/page/header/header.component.ts (hunk 8 above)
Comment (Abd-alRahim-Kojan) [reply src/app/shared/page/header/header.component.ts]: I add comment for that. same problem> I think it's conflict from Bootstrap
`<!-- Edit mt-3 class in ts file becucse it do conflict on the css code-->`

**Pull request merged**